src/nerdbank-streams/src/FrameHeader.ts

```diff
@@ -0,0 +1,43 @@
+import { ControlCode } from "./ControlCode";
+
+export class FrameHeader {
+    public static readonly HeaderLength = 1 /*control code*/ + 4 /*channel id*/ + 2/*payload length*/;
+
+    public static Deserialize(buffer: Buffer): FrameHeader {
+        if (!buffer || buffer.length !== FrameHeader.HeaderLength) {
+            throw new Error("buffer must have length of " + FrameHeader.HeaderLength);
+        }
+
+        const header = new FrameHeader();
+        header.code = buffer[0];
+        header.channelId = buffer.readUInt32BE(1);
+        header.framePayloadLength = buffer.readUInt16BE(5);
+        return header;
+    }
+
+    /**
+     * Gets or sets the kind of frame this is.
+     */
+    public code: ControlCode;
+
+    /**
+     * Gets or sets the channel that this frame refers to or carries a payload for.
+     */
+    public channelId: number;
+
+    /**
```

Comment:
might consider making this a property and checking its validity, since number is too broad